### PR TITLE
test: add test for unicode range test

### PR DESCRIPTION
**Pull request by HerringtonDarkholme**
part of #1594

## Summary by CodeRabbit

- **Tests**
	- Added new test cases for validating node positions with ASCII and Unicode characters.
	- Enhanced the `test_unicode` function to assert the line number of the node's range.

### CODE REVIEW

File: crates/core/src/node.rs [hunk 1]
```diff
@@ -697,4 +697,23 @@ if (a) {
...
+
+  #[test]
+  #[ignore = "TODO: fix column to be unicode character"]
+  fn test_unicode_pos() {
+    let root = Tsx.ast_grep("🦀");
+    let root = root.root();
+    let node = root.find("$A").expect("should exist");
+    assert_eq!(node.start_pos(), (0, 0));
+    assert_eq!(node.end_pos(), (0, 1));
+  }
```
[the remark this answers is not in the document]
Comment (HerringtonDarkholme) [reply crates/core/src/node.rs:718]: No, pos should calculate character count, not character width.
